test/converters.test.ts

```diff
@@ -358,3 +368,32 @@ test("object converter", async () => {
   const r3 = await converter.convert(null, {});
   expect(r3).toEqual({ value: null });
 });
+
+test("dynamic decimal converter", async () => {
+  const context = { options: { decimalPlaces: 0 } };
+
+  function currency() {
+    return converters.decimal(context => context.options);
+  }
+
+  const M = types.model("M", {
+    foo: types.string
+  });
+
+  const form = new Form(M, {
+    foo: new Field(currency())
+  });
+
+  const o = M.create({ foo: "3" });
+
+  const state = form.state(o, { context: context });
+  const field = state.field("foo");
+
+  await field.setRaw("3.141");
+  expect(field.raw).toEqual("3.141");
+  expect(field.value).toEqual("3"); // conversion error
```

Comment:
You say here 'conversion error', but it's not a conversion error?

Reply:
Ah, I understand now -- your choice of 'foo' being 3 to start with is very unfortunate. I am reading this as: there are no decimals, so we strip away any decimals we got. But instead it's a conversion error. I suggest you change the starting value of 'foo' to '4' and in addition check for 'field.error' here.